src/plot/svgMask.tsx

```diff
@@ -0,0 +1,139 @@
+import { SvgRect } from '@h5web/lib';
+import type { SVGProps } from 'react';
+import { Vector3 } from 'three';
+
+type Rect = [Vector3, Vector3];
+
+interface Props extends SVGProps<SVGPathElement> {
+    coords: Rect;
+    strokePosition?: 'inside' | 'outside';
+    strokeWidth?: number;
+    numModules: Vector3,
+    gapFraction: Vector3, 
+    missingSegments: number[]
+}
+
+function SvgMask(props: Props){
+    const [detectorUpper, detectorLower] = props.coords;
+    // check if clone is needed later on
+    const fulllength = detectorLower.clone().sub(detectorUpper);
+    const gaplength = fulllength.clone().multiply(props.gapFraction);
+    const moduleLength = 
+        fulllength.clone()
+        .sub(gaplength.clone()
+        .multiply(props.numModules.clone().subScalar(1)))
+        .divide(props.numModules);
```

Comment:
should: add more explaining comments - when I look at it the first time I am not sure what is the system in question these equations model